src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementBackedDocumentsService.java

```diff
@@ -136,21 +137,18 @@ public DocumentManagementBackedDocumentsService(
     }
 
     @Override
-    public void generateDefendantPinLetter(String externalId, String pin, String authorisation) {
+    public void generateDefendantPinLetter(String externalId, String authorisation) {
         Claim claim = claimService.getClaimByExternalId(externalId, authorisation);
-        final String fileName = buildDefendantLetterFileBaseName(claim.getReferenceNumber());
-        Optional<URI> claimDocument = claim.getClaimDocument(DEFENDANT_PIN_LETTER);
-        if (!claimDocument.isPresent()) {
-            try {
-                PDF defendantLetter = new PDF(fileName,
-                    defendantPinLetterPdfService.createPdf(claim, pin),
-                    DEFENDANT_PIN_LETTER);
-                uploadToDocumentManagement(defendantLetter, authorisation, claim);
-            } catch (Exception ex) {
-                logger.warn(String.format("unable to upload document %s into document management",
-                    fileName), ex);
-            }
+        if (claim.getDefendantId() != null) {
+            throw new ConflictException("Cannot generate pin letter.Claim has already been linked to defendant.");
         }
+        final String fileName = buildDefendantLetterFileBaseName(claim.getReferenceNumber());
+        String pin = userService.generatePin(claim.getClaimData().getDefendant().getName(), authorisation)
+            .getPin();
+        PDF defendantLetter = new PDF(fileName,
+            defendantPinLetterPdfService.createPdf(claim, pin),
+            DEFENDANT_PIN_LETTER);
+        uploadToDocumentManagement(defendantLetter, authorisation, claim);
```

Comment:
There is an oversight here. That's not all there is to do. We also need to link the defendant to the claim using the new PIN.

Reply:
AFAIK.. the defendant will have to input the details i.e the claimnumber and pin in the first-contact/start link and login with an account which will then create the link.

Reply:
In existing support endpoint we don't do much more than that which begs the question..how does it work :-) I have asked support people - it is not clear yet. I am told that the support controller that is currently being used works when it comes to regenerating PIN Letters, therefore the only explanation is that the same PIN must be returned each time (instead of  generating a new one). That would defeat our purpose a bit since we would end up regenerating and storing always the same PIN Letter document.